frank/plot.py

```diff
@@ -31,308 +31,205 @@
-        Width of bins in which data is binned (for legend labels)
-    xscale : Matplotlib axis scale, default = 'log'
-        Scale for x-axis
-    yscale : Matplotlib axis scale, default = 'linear'
-        Scale for y-axis
-    normalize_resid : bool, default = False
-        Whether to plot the residuals normalized pointwise (i.e., locally)
-        by the data amplitude
+        Baseline data coordinates
+    resid : array
+        (Observed - fitted) visibility amplitudes at baselines
+    ax : Matplotlib `~.axes.Axes` class
+        Axis on which to plot
     """
 
-    resid = obs - fit
-    if normalize_resid:
-        resid /= obs
-    rmse = (np.mean(resid**2))**.5
-
-    ax.plot(baselines, resid, c=c, marker=marker, ls=ls,
-            label=r'{:.0f} k$\lambda$ bins, RMSE {:.3f}'.format(binwidth/1e3,
-                                                                rmse)
-            )
+    ax.plot(baselines, resid, **kwargs)
 
-    ax.set_xlabel(r'Baseline [$\lambda$]')
-    if normalize_resid:
-        ax.set_ylabel('Normalized\nresidual')
-    else:
-        ax.set_ylabel('Residual [mJy]')
-    ax.set_xscale(xscale)
-    ax.set_yscale(yscale)
-    ax.legend()
+    ax.legend(loc='best')
 
-    if yscale == 'linear':
-        ax.axhline(0, c='c', ls='--', zorder=10)
-        ax.set_ylim(-2 * rmse, 2 * rmse)
+    ax.axhline(0, c='c', ls='--', zorder=10)
 
 
-def plot_vis_hist(bin_edges, bin_counts, bin_width, ax, c): # TODO: just use hist
+def plot_vis_hist(bins, counts, ax, **kwargs):
     r"""
     Plot a histogram of visibilities using a precomputed binning
 
     Parameters
     ----------
-    bin_edges : array
-        Sequential left and right edges (baselines) of each bin.
-        The assumed unit (for the x-label) is :math:`\lambda`
+    bins : array
+        Bin edges
     bin_counts : array
         Number of visibilities in each bin
-    bin_width : float
-        Constant width of the bins.
-        The assumed unit (for the x-label and legend) is :math:`\lambda`
-    ax : Matplotlib axis
+    ax : Matplotlib `~.axes.Axes` class
         Axis on which to plot
-    c : Matplotlib color, default = 'k'
-        Color of bins
     """
 
-    xs, ys = [], []
-    for l, r, y in zip(bin_edges[0], bin_edges[1], bin_counts):
-        xs.extend([l,r])
-        ys.extend([y,y])
-
-    ax.hlines(ys[::2], xmin=xs[::2], xmax=xs[1::2], color=c,
-              label=r'Obs., {:.0f} k$\lambda$ bins'.format(bin_width/1e3))
-    ax.fill_between(xs, ys, color=c, alpha=.5)
+    ax.hist(0.5*(bins[1:]+bins[:-1]), bins, weights=counts, alpha=.5, **kwargs)
 
-    ax.set_xlabel(r'Baseline [$\lambda$]')
-    ax.set_ylabel('n')
-    ax.set_xscale('log')
-    ax.set_yscale('log')
-    ax.legend()
+    ax.legend(loc='best')
 
 
-def plot_pwr_spec(baselines, pwr_spec, ax, c='#B123D7', ls='-', ylolim=None,
-                  xscale='log', yscale='log'
-                  ):
+def plot_pwr_spec(baselines, pwr_spec, ax, **kwargs):
```

Comment:
I don't think this should be a function.

Reply:
217da63 combines  `plot_vis_fit`, `plot_vis`, `plot_vis_resid`, and `plot_pwr_spec` into the single function `plot_vis_quantity`.